### ISSUE TITLE
Use postMessage to transfer side modules between workers

### ISSUE THREAD

**Issue by sbc100** [emscripten-core/emscripten#17078]
### Discussed in https://github.com/emscripten-core/emscripten/discussions/17034

<div type='discussions-op-text'>

<sup>Originally posted by **alexander-veselov** May 24, 2022</sup>
I understand that "SIDE_MODULE + USE_PTHREADS" is experimental. But I would still like to know if this is a known issue and if it can be fixed somehow.

We have a lot of side modules (about 350MB). USE_PTHREADS=1 and PTHREAD_POOL_SIZE=20. With this configuration, an "out of memory" error occurs. It seems that all side modules are loaded several times for each thread.

For example, if you set PTHREAD_POOL_SIZE=1, then the problem will disappear.
I know that when building without side modules, the problem will also disappear, but we need side modules.

Is it possible to make threads use shared memory and not load copies of dynamic libs several times?</div>

**Comment by kleisauke** [comment]
I think this was fixed with PR #19390.

**Comment by sbc100** [comment]
Yes, this is partly completely, but there are still some cases where module sharing is not possible, so I'd like to keep this open while we work to impove on that.
